src/applications/static-pages/wizard/index.js

```diff
@@ -61,9 +61,13 @@ export class Wizard extends React.Component {
 
   setPageState = (index, newState, nextPageName) => {
     let newHistory = set(`[${index}].state`, newState, this.state.pageHistory);
-
-    // If the next page is new, rewrite the future history
-    if (nextPageName) {
+    if (nextPageName === index && newHistory.length - 1 > index) {
+      // Remove previous page (needed to not show messaging when an invalid
+      // value is provided; e.g. for an invalid date, we don't want to show an
+      // inappropriate alert message)
+      newHistory.pop();
+    } else if (typeof nextPageName === 'string') {
+      // If the next page is new, rewrite the future history
```

Comment:
I'm confused on where `nextPageName`'s type becomes dynamic (a string, or something else.) Am I overlooking it?

Reply:
`nextPageName` should return a string or `null`. At one point in making changes for this PR, I was passing an index number, but since that isn't necessary anymore, I'll modify this check. Thanks for catching it!

Reply:
oops, I was wrong. `nextPageName` in the `bdd.jsx` file passes the step index (`1`) to make sure that the wizard doesn't render a next page, and stays on the passed index number. The reason this is done is because we don't want the wizard to show an incorrect alert (eligible or not eligible message) when the actual date that was enter is invalid.

Reply:
Grrr, actually now that I understand the wizard better, I'm going to revert these changes - but I'll add some JSDocs - and manipulate the `pageHistory` within the `bdd.jsx` file.

Reply:
Haha thanks @Mottie 😄